src/camera-helper.ts

```diff
@@ -0,0 +1,143 @@
+  private trackConstraints: MediaTrackSettings;
+
+  constructor() {
+    this.stream = null;
+    this.track = null;
+
+    this.trackConstraints = {};
+  }
+
+  async getCameras() {
+    let devices: MediaDeviceInfo[] = [];
+
+    if (constants.SUPPORTS_MEDIA_DEVICES) {
+      devices = await navigator.mediaDevices.enumerateDevices();
+      devices = devices.filter((device) => device.kind === 'videoinput');
+    }
+
+    return devices;
+  }
+
+  async takePhoto(video: HTMLVideoElement): Promise<Blob | null> {
+    if (!constants.SUPPORTS_MEDIA_DEVICES) {
+      return null;
+    }
+
+    if (constants.SUPPORTS_IMAGE_CAPTURE) {
+      const stream = video.srcObject;
+
+      if (!stream) {
+        return null;
+      }
+
+      const track = stream.getVideoTracks()[0];
+      const capture = new ImageCapture(track);
+      const settings: PhotoSettings = {};
+
+      return await capture.takePhoto(settings);
+    } else {
+      const canvas = document.createElement('canvas');
+      const context = canvas.getContext('2d')!;
+      canvas.width = video.videoWidth;
+      canvas.height = video.videoHeight;
+      context.drawImage(video, 0, 0);
+
+      return await canvasToBlob(canvas, constants.IMAGE_TYPE);
+    }
+  }
+
+  stopStream() {
+    if (this.stream) {
+      for (const track of this.stream.getVideoTracks()) {
+        track.stop();
+      }
+    }
+  }
+
+  async startStream(deviceId) {
+    this.stopStream();
+
+    (streamConstraints.video as MediaTrackConstraints).deviceId = deviceId;
+
+    const stream = await navigator.mediaDevices.getUserMedia(streamConstraints);
+    this.stream = stream;
+    this.track = stream.getVideoTracks()[0];
+    this.trackConstraints = {};
+    return stream;
+  }
+
+  getSettings(): MediaTrackSettings {
+    if (!this.track || !this.track.getSettings) {
+      return {};
+    }
+    return this.track.getSettings();
+  }
+
+  getCapabilities(): MediaTrackCapabilities {
+    if (!this.track || !this.track.getCapabilities) {
+      return {};
+    }
+    return this.track.getCapabilities();
+  }
+
+  setConstraint(name: string, value: any) {
+    if (name in supportedConstraints && supportedConstraints[name]) {
+      const capabilities = this.getCapabilities();
+      console.log(capabilities[name], name, value);
+      if (capabilities[name]) {
+        this.trackConstraints[name] = value;
+      }
+    }
+    this.applyConstraints();
+  }
+
+  applyConstraints() {
+    if (this.track && this.track.applyConstraints) {
+      const constraints = this.track.getConstraints();
+      const advanced: MediaTrackConstraintSet[] = constraints.advanced || [];
+
+      for (const [name, value] of Object.entries(this.trackConstraints)) {
+        const constraint = {};
+        constraint[name] = value;
+        advanced.push(constraint);
+      }
+
+      if (advanced.length > 0) {
+        constraints.advanced = advanced;
+      }
+
+      this.track.applyConstraints(constraints);
```

Comment:
Same here: How about `return await this.track.applyConstraints(constraints);` since it returns a promise?